src/datasets/download/streaming_download_manager.py

```diff
@@ -759,7 +759,7 @@ def _iter_from_urlpaths(
                 yield urlpath
             else:
                 for dirpath, dirnames, filenames in xwalk(urlpath, use_auth_token=use_auth_token):
-                    for i, dirname in enumerate(dirnames):
+                    for i, dirname in enumerate(dirnames[:]):
```

Comment:
Why is this needed ? Is it related to the `del dirnames[i]` call ? If so maybe it's simpler to just remove `del dirnames[i]` instead. I feel like this optimization adds more confusion that it brings value

Reply:
Yes. There is a comment that explains this inside the loop... It's outside of it now. And I removed the `[:]` syntax.